src/sql/AtlasSQLite.cpp

```diff
@@ -0,0 +1,90 @@
+#include "AtlasSQLite.h"
+
+namespace atlas {
+
+AtlasSQLite::AtlasSQLite()
+{
+	bConnected = false;
+}
+
+AtlasSQLite::~AtlasSQLite()
+{
+	closeConnection();
+	fprintf(stdout, "Connection closed\n");
+}
+
+void AtlasSQLite::closeConnection()
+{
+	if(pCon)
+		sqlite3_close(pCon);
+}
+
+bool AtlasSQLite::isConnected()
+{
+    return bConnected;
+}
+
+bool AtlasSQLite::openConnection(std::string databaseName, std::string databaseDir)
+{
+	SQLiteDatabaseName = databaseName;
+	SQLiteDBPath	   = databaseDir;
+	
+	bConnected = true;
+
+	std::string db=SQLiteDatabaseName;
+	std::string dir=SQLiteDBPath;
+	std::string path=dir.append(db);
+
+	int rc = sqlite3_open(path.c_str(), &pCon);
+	
+	if(rc)
+	{
+        fprintf(stderr, "Can't open database: %s\n", sqlite3_errmsg(pCon));
+	    bConnected = false;
+	}
+
+    /* Create table 'IdentityPSK' if not exists */
+
+    sql = "CREATE TABLE IF NOT EXISTS IdentityPSK("  \
+        "Id INTEGER PRIMARY KEY     AUTOINCREMENT," \
+        "Identity           TEXT    NOT NULL," \
+        "PSK            TEXT     NOT NULL );";
+
+    rc = sqlite3_exec(pCon, sql, NULL, 0, &lastError);
+    
+    if( rc != SQLITE_OK )
+    {
+        fprintf(stderr, "SQL error: %s\n", lastError);
+        sqlite3_free(lastError);
+		bConnected = false;
+    } 
+
+	fprintf(stdout, "Connection opened\n");
+
+	return bConnected;
+}
+uint16_t AtlasSQLite::insert(std::string identity, std::string psk)
+{
+    if(!isConnected())
+		return 0;
+
+	std::string buf("INSERT INTO IdentityPSK(Identity,PSK)VALUES(");
```

Comment:
What if identity already exists? Identity should be marked as unique in the db schema and an update should be performed in this case.